frontend/components/product-list/sections/FilterableProductsSection/RangeInput.tsx

```diff
@@ -0,0 +1,215 @@
+   });
+
+   const max = useDebouncedValue({
+      currentAlgoliaValue: appliedRangeMax,
+      refine: refineMaxValue,
+   });
+
+   const isValidRange =
+      min.valueAsNumber == null ||
+      max.valueAsNumber == null ||
+      min.valueAsNumber < max.valueAsNumber;
+
+   const minInputId = `${props.attribute}-input-min`;
+   const maxInputId = `${props.attribute}-input-max`;
+   const formattedAttribute = formatFacetName(props.attribute).toLowerCase();
+
+   return (
+      <FormControl isInvalid={!isValidRange}>
+         <HStack>
+            <Box>
+               <FormLabel htmlFor={minInputId} srOnly>
+                  Set min {formattedAttribute}
+               </FormLabel>
+               <InputGroup size="sm">
+                  <InputLeftElement pointerEvents="none" color="gray.500">
+                     $
+                  </InputLeftElement>
+                  <Input
+                     type="number"
+                     id={minInputId}
+                     placeholder={String(range.range.min)}
+                     value={min.valueAsText}
+                     onChange={(event) => {
+                        const value = event.target.value;
+                        min.setValueText(value);
+                     }}
+                     borderRadius="md"
+                  />
+               </InputGroup>
+            </Box>
+            <Text>to</Text>
+            <Box>
+               <FormLabel htmlFor={maxInputId} srOnly>
+                  Set max {formattedAttribute}
+               </FormLabel>
+               <InputGroup size="sm">
+                  <InputLeftElement pointerEvents="none" color="gray.500">
+                     $
+                  </InputLeftElement>
+                  <Input
+                     type="number"
+                     id={maxInputId}
+                     placeholder={String(range.range.max)}
+                     value={max.valueAsText}
+                     onChange={(event) => {
+                        const value = event.target.value;
+                        max.setValueText(value);
+                     }}
+                     borderRadius="md"
+                  />
+               </InputGroup>
+            </Box>
+         </HStack>
+         {!isValidRange && (
+            <FormErrorMessage data-test="input-error">
+               max should be higher than min
+            </FormErrorMessage>
+         )}
+      </FormControl>
+   );
+}
+
+function parseNumber(text: string) {
+   const number = parseFloat(text);
+   return Number.isFinite(number) ? number : undefined;
+}
+
+function clamp(
+   value: number | undefined,
+   min: number | undefined,
+   max: number | undefined
+) {
+   return value == null
+      ? value
+      : Math.min(Math.max(value, min ?? -Infinity), max ?? Infinity);
+}
+
+type UseDebouncedValueProps = {
+   refine: (value: number | undefined) => void;
+   currentAlgoliaValue: number | undefined;
+};
+
+function useDebouncedValue({
```

Comment:
It would be useful to have a function doc here explaining the feature this adds on top of `useDebounce()`

Reply:
This function is specific of this component, it basically decouples the input from the Algolia state, so that it doesn't feel laggy. I'll add the explanation in the code 👍